src/module/token.ts

```diff
@@ -34,22 +34,30 @@ export class LancerTokenDocument extends TokenDocument {
 export class LancerToken extends Token {
   constructor(document: LancerTokenDocument) {
     super(document);
-    this._spaces = [];
+    this._spaces = {
+      at: { x: -1, y: -1 },
+      spaces: [],
+    };
   }
 
   /**
    * Cached occupied spaces
    */
-  protected _spaces: Point[];
+  protected _spaces: { at: Point; spaces: Point[] };
 
   /**
    * Returns a Set of Points corresponding to the grid space center points that
    * the token occupies.
    */
   getOccupiedSpaces(mode: "current position" | "updated position" = "current position"): Point[] {
-    let pos: { x: number, y: number } = mode == "current position" ? this.position : this.data;
+    let pos: { x: number; y: number } = mode == "current position" ? this.position : this.data;
+    // Invalidate the cache if the position is different than when it was last calculated.
+    if (Math.floor(pos.x) !== Math.floor(this._spaces.at.x) || Math.floor(pos.y) !== Math.floor(this._spaces.at.y)) {
```

Comment:
It concerns me that we change what we're looking at to invalidate the cache depending on what the `mode` sent in is. It's probably fine, overall — in any case where something requests the updated position, that position probably _is_ the current position on a later call, but it makes me uncomfortable.

I'm not requesting any code changes here, but can we add like a rust-style "this unsafe block is safe, we think, because" comment?

Reply:
Even if a later call requests the current position, the result will be the same if the coordinate for pos is the same. I'm not sure what the unsafe condition is here.